spec/element_base_spec.rb

```diff
@@ -293,6 +305,12 @@
     end
   end
 
+  describe '#titles' do
+    it 'returns elements with data-type title' do
+      expect(title_book.titles.to_s).to eq('<div data-type="title">This is a title</div>')
```

Comment:
Can you put two titles in the `title_book`, one that is a `div` and one that is a `span`?  

```
<div data-type="title">One</div>
<span data-type="title">Two</span>
```

And then have your test show that the iterator gets both?  

```ruby
expect(title_book.titles.map(&:text)).to eq(["One", "Two"])
```